library/src/animationManager.cxx

```diff
@@ -304,38 +263,92 @@ int animationManager::GetAnimationIndex()
 std::string animationManager::GetAnimationName()
 {
   assert(this->Importer);
-  if (this->AvailAnimations <= 0)
+  if (this->AvailAnimations <= 0 || this->AnimationIndex == -2)
   {
     return "No animation";
   }
 
-  if (this->AnimationIndex < 0)
+  if (this->AnimationIndex == -1)
   {
     return "All Animations";
   }
   return this->Importer->GetAnimationName(this->AnimationIndex);
 }
 
 //----------------------------------------------------------------------------
-void animationManager::EnableOnlyCurrentAnimation()
+void animationManager::UpdateForAnimationIndex()
 {
   assert(this->Importer);
+
+  if (this->AnimationIndex == this->Options.scene.animation.index || this->AvailAnimations <= 0)
+  {
+    // Already updated or no animation available
+    return;
+  }
+
+  this->AnimationIndex = this->Options.scene.animation.index;
+  if (this->AnimationIndex > 0 && this->AnimationIndex >= this->AvailAnimations)
+  {
+    log::warn(
+      "Specified animation index is greater than the highest possible animation index, enabling "
+      "the first animation.");
+    this->AnimationIndex = 0;
+  }
+
   for (int i = 0; i < this->AvailAnimations; i++)
   {
     this->Importer->DisableAnimation(i);
   }
   for (int i = 0; i < this->AvailAnimations; i++)
   {
-    if (this->AnimationIndex < 0 || i == this->AnimationIndex)
+    if (this->AnimationIndex == -1 || i == this->AnimationIndex)
     {
       this->Importer->EnableAnimation(i);
     }
   }
+
+  // Display currently selected animation
+  log::debug("Current animation is: ", this->GetAnimationName());
+
+  // Recover time ranges for all enabled animations
+  this->TimeRange[0] = std::numeric_limits<double>::infinity();
+  this->TimeRange[1] = -std::numeric_limits<double>::infinity();
+  for (vtkIdType animIndex = 0; animIndex < this->AvailAnimations; animIndex++)
+  {
+    if (this->Importer->IsAnimationEnabled(animIndex))
+    {
+      double timeRange[2];
+      int nbTimeSteps;
+      vtkNew<vtkDoubleArray> timeSteps;
+
+      // Discard timesteps, F3D only cares about elapsed time using time range and deltaTime
+      // Specifying a frame rate (60) in the next call is not needed after VTK 9.2.20230603 :
+      // VTK_VERSION_CHECK(9, 2, 20230603)
+      this->Importer->GetTemporalInformation(animIndex, 60, nbTimeSteps, timeRange, timeSteps);
+
+      // Accumulate time ranges
+      this->TimeRange[0] = std::min(timeRange[0], this->TimeRange[0]);
+      this->TimeRange[1] = std::max(timeRange[1], this->TimeRange[1]);
+    }
+  }
+  if (this->TimeRange[0] > this->TimeRange[1])
+  {
+    log::warn("Animation(s) time range delta is invalid: [", this->TimeRange[0], ", ",
+      this->TimeRange[1], "]. Swapping range.");
+    std::swap(this->TimeRange[0], this->TimeRange[1]);
```

Comment:
There is no way to cover this at the moment, at least without editing some binary exodus files, which I will not do.